scripts/upload_results_to_oep.py

```diff
@@ -0,0 +1,178 @@
+
+
+oem2orm = Mock()
+
+SCHEMA = "model_draft"
+
+
+def load_json_to_dict(filepath):
+    with open(filepath, "rb") as f:
+        return json.load(f)
+
+
+def save_dict_to_json(data, filepath, encoding="utf-8"):
+    with open(filepath, "w", encoding=encoding) as f:
+        return json.dump(data, f, sort_keys=True, indent=2)
+
+
+metadata_folder = pathlib.Path("oemof_b3/schema/oemetadata.json")
+oemetadata_template = load_json_to_dict(metadata_folder)
+
+
+def create_metadata(data, template=None):
+    if template is None:
+        template = oemetadata_template
+
+    metadata = template.copy()
+
+    return metadata
+
+
+if __name__ == "__main__":
+    filepath = pathlib.Path(sys.argv[1])
+    metadata_path = pathlib.Path(sys.argv[2])
+    logfile = sys.argv[3]
+
+    # set up the logger
+    scenario = "scenario"  # TODO: Derive scenario from filepath
+    logger = config.add_snake_logger(logfile, "upload_results_to_oep")
+
+    if not os.path.exists(metadata_path):
+        os.makedirs(metadata_path)
+
+    # find data to upload
+    dict_table_filename = {
+        os.path.splitext(filename)[0]: filename for filename in os.listdir(filepath)
+    }
+    logger.info(
+        "These files will be uploaded: " + ", ".join(dict_table_filename.values())
+    )
+
+    # Setting up the oem2orm logger
+    # If you want to see detailed runtime information on oem2orm functions or if errors occur,
+    # you can activate the logger with this simple setup function.
+    # oem2orm.setup_logger()
+
+    # To connect to the OEP you need your OEP Token and user name. Note: You ca view your token
+    # on your OEP profile page after
+    # [logging in](https://openenergy-platform.org/user/login/?next=/).
+    # The following command will prompt you for your token and store it as an environment variable.
+    # When you paste it here, it will only show dots instead of the actual string.
+    # save user name & token in environment as OEP_TOKEN & OEP_USER
+    db = oem2orm.setup_db_connection()
+
+    # Create the metadata and save it
+    for table, filename in dict_table_filename.items():
+        data_upload_df = pd.read_csv(
+            os.path.join(filepath, filename), encoding="utf8", sep=";"
+        )
+
+        metadata = create_metadata(data_upload_df)
+
+        save_dict_to_json(metadata, metadata_path / f"{table}.json")
+
+        logger.info(f"Saved metadata to: {metadata_path}")
+
+    # Creating sql tables from oemetadata
+    metadata_folder = oem2orm.select_oem_dir(oem_folder_name=metadata_path)
+
+    # The next command will set up the tables. The collect_tables-function collects all metadata
+    # files in a folder, creates the SQLAlchemy ORM objects and returns them. The tables are
+    # ordered by foreign key. Having a valid metadata strings is necessary for the following steps.
+    tables_orm = oem2orm.collect_tables_from_oem(db, metadata_folder)
+
+    # create tables
+    oem2orm.create_tables(db, tables_orm)
```

Comment:
@jh-RLI: Can these two steps be performed for a single metadata.json, one by one? If yes, it could be part of the loop over the files.